### PR TITLE
add `reportUnreachable` option

### LINKED ISSUE #11: option to ban unreachable code
unreachable code is very bad because it's not typechecked at all. related: #7

### PR DESCRIPTION
**Pull request by DetachHead**
fixes #11

partially addresses #7. need to figure out why some unreachable code gets checked now but not others

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -373,6 +379,7 @@ The following table lists the default severity levels for each diagnostic rule w
 | reportUninitializedInstanceVariable       | "none"     | "none"     | "none"     | "none"     |
 | reportUnnecessaryTypeIgnoreComment        | "none"     | "none"     | "none"     | "none"     |
 | reportUnusedCallResult                    | "none"     | "none"     | "none"     | "none"     |
+| reportUnreachable                         | "none"     | "none"     | "none"     | "none"     |
```
Comment (KotlinIsland) [inline docs/configuration.md:382]: it's enabled in strict

File: packages/pyright-internal/src/tests/samples/unreachable2.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from typing import TYPE_CHECKING
+
+if not TYPE_CHECKING:
+    ...
+
+if TYPE_CHECKING:
+    ...
+else:
+    ...
```
Comment (KotlinIsland) [inline packages/pyright-internal/src/tests/samples/unreachable2.py:9]: might be nice to have an additional 
```py
if TYPE_CHECKING:
    ...
else:
    ...
```

**Review by KotlinIsland — APPROVED** [review]

File: docs/configuration.md (hunk 1 above)
Comment (DetachHead) [reply docs/configuration.md:382]: changed it to disabled by default. #1 will address this

File: packages/pyright-internal/src/tests/samples/unreachable2.py (hunk 2 above)
Comment (DetachHead) [reply packages/pyright-internal/src/tests/samples/unreachable2.py:9]: the `if not TYPE_CHECKING` above reproduces the same scenario

**Pull request merged**